cmake/compiler/gcc/target_security_fortify.cmake

```diff
@@ -5,9 +5,13 @@ macro(toolchain_cc_security_fortify)
     # _FORTIFY_SOURCE: Detect common-case buffer overflows for certain functions
     # _FORTIFY_SOURCE=1 : Compile-time checks (requires -O1 at least)
     # _FORTIFY_SOURCE=2 : Additional lightweight run-time checks
-    zephyr_compile_definitions(
-      _FORTIFY_SOURCE=2
-    )
+
+    if(NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
+      # clang errors on _FORTIFY_SOURCE=2
+      zephyr_compile_definitions(
+        _FORTIFY_SOURCE=2
+	)
+    endif()
```

Comment:
This isn't sane. We are doing "if clang" in a gcc file. Clang should instead not use gcc's target_security_fortify.cmake.

Reply:
@SebastianBoe 

You are right. I checked, clang doesn't understand fortify at all.  I refined it, please take a look.